Encryption.md

```diff
@@ -75,55 +75,126 @@ written to the output stream.
 
 Parquet encryption algorithms are based on the standard AES ciphers for symmetric 
 encryption. AES is supported in Intel and other CPUs with hardware acceleration of 
-crypto operations (“AES-NI”) - that can be leveraged by e.g. Java programs 
+crypto operations (“AES-NI”) - that can be leveraged, for example, by Java programs 
 (automatically via HotSpot), or C++ programs (via EVP-* functions in OpenSSL).
 
-Initially, two algorithms are implemented, one based on a GCM mode of AES, and the other 
+Initially, two algorithms have been implemented, one based on a GCM mode of AES, and the other 
 on a combination of GCM and CTR modes.
 
 AES-GCM is an authenticated encryption. Besides the data confidentiality (encryption), it 
-supports two levels of integrity verification / authentication: of the data (default), and 
+supports two levels of integrity verification (authentication): of the data (default), and 
 of the data combined with an optional AAD (“additional authenticated data”). The default 
 authentication allows to make sure the data has not been tampered with. An AAD is a free 
 text to be signed, together with the data. The user can, for example, pass the file name 
-with its version (or creation timestamp) as the AAD, to verify the file has not been 
-replaced with an older version.
+with its version (or creation timestamp) as the AAD, to verify that the file has not been 
+replaced with an older version (see below).
```

Comment:
See what below? Which section? This should be a description, ideally with a link.

Reply:
In the AAD section below, an AAD scheme is specified. If an AAD scheme is required, then the statement that the user can do something is false. This encryption spec requires that the AAD is used to authenticate streams such that no stream can be replaced. with one from a different file or moved.

Reply:
From below:

> Parquet  modular encryption uses the AAD mode to protect against swapping ciphertext modules inside a file, between files - or against swapping full files (for example, replacement of a file with an old version)"